lib/spack/spack/build_systems/autotools.py

```diff
@@ -415,6 +417,13 @@ def configure(self, spec, prefix):
         with working_dir(self.build_directory, create=True):
             inspect.getmodule(self).configure(*options)
 
+    def setup_build_environment(self, env):
+        if (self.spec.platform == 'darwin'
+                and macos_version() >= Version('11')):
+            # Many configure files rely on matching '10.*' for macOS version
+            # detection and fail to add flags if it shows as version 11.
+            env.set('MACOSX_DEPLOYMENT_TARGET', '10.16')
```

Comment:
I think it would be good if we could configure this in a Spack config.

There will be requirements for both local architectures (e.g., `M1` is 11.0+) and C++ features (e.g., C++17 is 10.4+, C++20/23 to be looked up, etc.).

Reply:
I'm open to suggestions -- this was a quick fix to get the majority of packages working. The best way is probably to make sure `libtool` is fixed and up-to-date and always autoreconf on new macos systems.